client/command/generate.go

```diff
@@ -42,6 +42,7 @@ import (
 	"github.com/bishopfox/sliver/protobuf/clientpb"
 	"github.com/bishopfox/sliver/protobuf/commonpb"
 	"github.com/bishopfox/sliver/protobuf/rpcpb"
+	"github.com/bishopfox/sliver/server/db"
```

Comment:
This is a no-go: the `client` package can **never** import the `server` package.
If you need to retrieve information from the server (or the database), please use or create a dedicated RPC.

Reply:
Hrm so I use the same function, `GenerateUniqueIP()`,  in  `/server/c2/tcp-wg.go` during the key exchange. If that becomes an RPC, is it fine if it is called from within the server itself? I can't find anywhere in the codebase where this happens so i'm guessing no, in which case, what would be the best way to implement this?

Reply:
The main reason we don't import the `server` package in the `client` package is to avoid a dependency loop. Some parts of the server already import the `client` package, so doing the reverse isn't going to end well.

You can call stuff from `server/c2` in the server code without issue.

Reply:
Ok so i ended up moving the `generateUniqueIP()` code into `server/generate/` which is used by server code; additionally, there is the `GenerateUniqueIP` RPC, which wraps the previous function, and is used by the client. Let me me know if you're happy with that or want it changed. Changes [here](https://github.com/BishopFox/sliver/pull/363/commits/f0e0bfdc2baaf0ddc605aa883f4dc8950ca81341)